src/Makefile_Ecore.am

```diff
@@ -85,6 +86,8 @@ lib/ecore/efl_io_file.c \
 lib/ecore/efl_io_copier.c \
 lib/ecore/efl_io_buffered_stream.c \
 lib/ecore/efl_promise.c \
+lib/ecore/efl_promise2.c \
+lib/ecore/efl_promise2_priv.c \
```

Comment:
no need to split this into multiple files... just add the `Eina_Value` stuff at the end.